### DISCUSSION TITLE
updating window in jupyter with ipywidget

### DISCUSSION THREAD

**Discussion by julienguegan** [pyvista/pyvista#2239] [Q&A] [answered]
Hi,

I wanted to know if there is a way to update an ipyvista window using value of ipywidget in jupyter-notebook. Currently, I have to run another cell and call `plotter.show()` each time I want to update something ...

Thank you !

**Comment by banesullivan** [comment, marked as answer]
This is possible with `ipyvtklink` and PyVista: https://docs.pyvista.org/user-guide/jupyter/ipyvtk_plotting.html

```py
import pyvista as pv
import ipywidgets

sphere = pv.Sphere()

plotter = pv.Plotter(notebook=True)
actor = plotter.add_mesh(sphere)
viewer = plotter.show(jupyter_backend='ipyvtklink', return_viewer=True)

def callback(*args):
    actor.SetVisibility(not actor.GetVisibility())
    viewer.update_canvas()

button = ipywidgets.ToggleButton(
    value=True,
    description='Visibility',
    icon='eye' # (FontAwesome names without the `fa-` prefix)
)
button.observe(callback )
button
```

```py
viewer
```

https://user-images.githubusercontent.com/22067021/176053741-a8bca05a-80da-4956-a9e7-e137f2496006.mov
